Overtake grunt.config properties to spawned childs

JanST123 opened the pull request:
Sometimes you want some properties from grunt.config to persist in the spawned task. To do this you can put config: <Array of config props> in your tasks configuration, and grunt-parallel will apply the given properties to the spawned grunt process.

JanST123 committed:
overtake some config properties to spawned task

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update README.md

amakhrov commented on tasks/parallel.js:
```diff
@@ -52,6 +65,19 @@ module.exports = function(grunt) {
+            configObj[configProp] = grunt.config.get(configProp);
+        });
+        
+        flags.push('--parallelconfig=' + JSON.stringify(configObj));
```
what if the stringified `configObj` contains whitespaces? will it still be parsed correctly?

JanST123 replied:
Yeah it should work, as there is some escaping mechanism for the args in grunt.spawn...

amakhrov replied:
Well, grunt itself doesn't do any extra escaping (https://github.com/gruntjs/grunt-legacy-util/blob/master/index.js#L185)
Perhaps Node.js does it, though.

Could you please test that to make sure it's supported nicely?

JanST123 replied:
Tested and works :-)
```javascript
space_test: "foo bar",
        space_test_1: { test: "" },
        
        parallel: {
            space_test: {
                options: {
                    grunt: true,
                    options: ['space_test']
                },
                tasks: ['space_test_1:test']
            }
}

grunt.registerMultiTask('space_test_1', 'space test', function() {
        grunt.log.write("value of config " + grunt.config.get('space_test'));
});
```

**Output:**
```
Jans-MBP:Frontend jan$ grunt parallel:space_test
Running "parallel:space_test" (parallel) task

    Running "space_test_1:test" (space_test_1) task
    value of config foo bar
    Done, without errors.
    
Done, without errors.
```

amakhrov replied:
Thanks!

amakhrov reviewed (commented):
Some further thoughts:

- I feel `options: { options: [] }` can be somewhat confusing. I would suggest renaming the new option to smth more specific to the use case, like `exposeGruntConfigKeys`.
- Not sure I exactly understand how it works. In the code you refer to `options.config`. But in the readme you have an example with `options.options`. Could you please clarify that?
- Could you please provide a more real-world use case for this feature? The one in the updated readme doesn't make it clear how passing config options this way is beneficial at all (or maybe it's just me). Also the  'space test' example you provided works just fine without any changes to `grunt-parallel`.
- For the final version of PR could you please make sure the codestyle is consistent with the rest of the codebase (spaces around ` = `, no spaces inside `function(args)`). I know ideally it's all identified by tools like eslint - but we don't have this set up for this project unfortunately.

Commit:
Update parallel.js

Commit:
Update README.md

JanST123 commented:
You're right and also I messed up the property name in the README.
I renamed it to ``exposeGruntConfigKeys`` like you suggested, and also fixed the README and provided a working real-world example.
I also fixed the coding style.

amakhrov commented on tasks/parallel.js:
```diff
@@ -9,6 +9,18 @@
+  if (config) {
+        // merge back in config properties
+        try {
+            var configObj=JSON.parse(config);
```
spaces please :)

amakhrov commented on README.md:
```diff
@@ -119,6 +119,55 @@ grunt.initConfig({
+                                        'git', 
+                                        ["rev-parse", "HEAD"]
+                                     )).split("\n")[0]);
+});
```
There is smth wrong with indentation across this whole code snippet (e.g line 145 indented with 4 spaces, line 161 - 7 spaces).
Could you please update it to have consistent indentation/spacing?

amakhrov commented on README.md:
```diff
@@ -119,6 +119,55 @@ grunt.initConfig({
 
+#### Expose grunt.config properties
+
+Sometimes you want some properties from grunt.config to persist in the spawned task. To do this you can put `exposeGruntConfigKeys: <Array of config props>` in your tasks configuration, and grunt-parallel will apply the given properties to the spawned grunt process.
```
Thanks for providing a more realistic example. I think now I finally understand the use case :) .
Seem that the main part here is that config values can be changed dynamically (by other tasks) - and special handling is needed for that. Constant config values are always available to child processes as is, without any extra configuration.
I feel it's important enough to be explicitly mentioned in the README.
What do you think?

amakhrov reviewed (commented):
Thanks for the updates @JanST123 . The new use case example makes much more sense to me know.

Commit:
Update README.md

Commit:
Update README.md

Commit:
Update parallel.js

JanST123 commented:
Morning @amakhrov 

I have now pointed out the main reason for this feature in the README. What do you think?

Also fixed indentation and missing spaces ;)

amakhrov commented:
@JanST123 everything looks great. Sorry, I'm a bit behind my schedule this week. I'll try to merge and release a new version soon.
